Flickr/Flickr/ImageCollectionViewController.swift

```diff
@@ -10,17 +10,134 @@
+        static let baseURLString = Constants.FlickrAPI.baseUrl
+        static let perPage = 100
+        // MARK: URLRequestConvertible
+        func asURLRequest() throws -> URLRequest {
+            let result: (path: String, parameters: Parameters) = {
+                switch self {
+                case let .search(text, page) :
+                    var searchParams = Constants.searchParams
+                    searchParams["text"] = text
+                    searchParams["page"] = page
+                    return (Constants.FlickrAPI.path, searchParams)
+                case let .popular(page):
+                    var popularParams = Constants.popularParams
+                    popularParams["page"] = page
+                    return  (Constants.FlickrAPI.path, popularParams)
+                }
+            }()
+            let url = try Router.baseURLString.asURL()
+            let urlRequest = URLRequest(url: url.appendingPathComponent(result.path))
+            return try URLEncoding.default.encode(urlRequest, with: result.parameters)
+        }
+    }
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        searchCollectionView.alpha = 0
+        popularCollectionView.alpha = 1
+        recentTableView.alpha = 0
+        searchConstraint.priority = UILayoutPriority(rawValue: 999)
+        searchConstraint.isActive = true
+        searchTextField.delegate = self
+        definesPresentationContext = true
+        recentTableView.delegate = self
+        recentTableView.dataSource = self
+        searchTextField.addTarget(self, action: #selector(clickOnTextEventFunc), for: UIControl.Event.touchDown)
+        searchTextField.addTarget(self, action: #selector(editingTextEventFunc), for: UIControl.Event.editingChanged)
+        recentTableView.rowHeight = UITableView.automaticDimension
+        recentTableView.register(UINib.init(nibName: "RecentTableViewCell", bundle: nil), forCellReuseIdentifier: "RecentCell")
+        searchTextField.attributedPlaceholder = NSAttributedString(string: "Search Flickr",
+                                                                   attributes: [NSAttributedString.Key.foregroundColor: UIColor.white])
     }
     
-    @IBOutlet weak var collectionView: UICollectionView!
-    @IBOutlet weak var searchTextField: UITextField!
-    @IBAction func searchButtonAction(_ sender: UIButton) {
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        rebuildTableSize()
+        performFlickrPopular()
+    }
+    override func viewDidLayoutSubviews(){
+        rebuildTableSize()
+        recentTableView.reloadData()
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {     // frozen for a while
+        if let identifier = segue.identifier {
+            if identifier == Constants.SegueIdentifier.GallerySegue,
+                let gvcvc = segue.destination as? ImageDetailViewController,
```

Comment:
gvcvc - gallery view controller view controller ???
need proper naming for this